### DISCUSSION TITLE
Unable to preventDefault inside passive event listener invocation

### DISCUSSION THREAD

**Discussion by boro0505** [idiotWu/smooth-scrollbar#459] [Q&A]
Hi Team, I keep getting this error again when combining with angular 13.
![image](https://user-images.githubusercontent.com/9208853/158145942-91c88e64-5fa7-42ce-8ff5-ce8ad11b5fd8.png)

Smooth scrollbar version, I use the latest one.
![image](https://user-images.githubusercontent.com/9208853/158146140-38cd6a96-00fe-4031-8ead-81587ed7991d.png)
Is there any way to fix this error?
Thanks TEAM.

**Comment by dolphin-wood** [comment]
That is a bit wired as we already pass `passive: false` to event handlers:
https://github.com/idiotWu/smooth-scrollbar/blob/5e86157f2a23a5513b097f21842397fd876d4f0e/src/utils/event-hub.ts#L35

Is anyone else facing this issue?

**Comment by boro0505** [comment]
@idiotWu 
So how do I deal with them? It always shows an error every time I scroll.

**Reply by dolphin-wood** [reply]
No idea here. Let's see if someone could help.

**Reply by dolphin-wood** [reply]
What's the browser you are using? Is this issue also happening in other browsers?

**Reply by boro0505** [reply]
i'm using chrome, it seems to be on chrome only.

**Reply by dolphin-wood** [reply]
What's the version of your Chrome & OS? Does it also happen on the demo site?

I can't reproduce it with Chrome 99.0.4844.51 on macOS 12.2.1 :/.

**Reply by boro0505** [reply]
Macos Mojave 10.14.5 and Chorme for Version 99.0.4844.51 (Official Build) (x86_64)
I only have local, no demo on the website. @idiotWu

**Reply by dolphin-wood** [reply]
I meant the demo page for this project: https://idiotwu.github.io/smooth-scrollbar/.
Is it also happening?

**Reply by boro0505** [reply]
no, Your demo page is normal.

**Comment by dolphin-wood** [comment]
It seems this issue is caused by `zone.js`, see also: https://github.com/angular/angular/issues/45020

**Reply by boro0505** [reply]
![image](https://user-images.githubusercontent.com/9208853/158161415-1b422418-e5b8-4471-bcd5-707ab4bad8ec.png)
i got error since i started to install smooth scrollbar.

**Reply by dolphin-wood** [reply]
`zone.js` patched the global `addEventListener` and therefore broke this plugin. I'm afraid there's little thing we can do in this plugin 🤔.

**Reply by boro0505** [reply]
@idiotWu 
can you update the plugin patch? I think it is essential for current angular versions.

**Reply by dolphin-wood** [reply]
I don't think it's easy to patch a fix in smooth-scrollbar. If we arbitrarily pass `{ passive: false }` as the third option to `addEventListener()`, it will be recognized as `capture: true` in old browsers that don't support event listener options and will break this plugin (especially the "continuously scrolling").

Anyway, I think this is an issue for angular and I'm not an angular user :/. I'd suggest you [file an issue in angular](https://github.com/angular/angular/issues) for further help.

**Reply by boro0505** [reply]
Thanks for your support.
